perfkitbenchmarker/providers/gcp/gce_network.py

```diff
@@ -121,15 +124,87 @@ def DisallowAllPorts(self):
       firewall_rule.Delete()
 
 
+class GceNetworkSpec(network.BaseNetworkSpec):
+
+  def __init__(self, project=None, **kwargs):
+    """Initializes the GceNetworkSpec.
+
+    Args:
+      project: The project for which the Network should be created.
+      kwargs: Additional key word arguments passed to BaseNetworkSpec.
+    """
+    super(GceNetworkSpec, self).__init__(**kwargs)
+    self.project = project
+
+
+class GceNetworkResource(resource.BaseResource):
+  """Object representing a GCE Network resource."""
+
+  def __init__(self, name, project):
+    super(GceNetworkResource, self).__init__()
+    self.name = name
+    self.project = project
+
+  def _Create(self):
+    """Creates the Network resource."""
+    create_cmd = [FLAGS.gcloud_path,
+                  'compute',
+                  'networks',
+                  'create',
+                  '--range=10.0.0.0/16',
+                  self.name]
+    create_cmd.extend(util.GetDefaultGcloudFlags(self))
+    vm_util.IssueCommand(create_cmd)
+
+  def _Delete(self):
+    """Deletes the Network resource."""
+    delete_cmd = [FLAGS.gcloud_path,
+                  'compute',
+                  'networks',
+                  'delete',
+                  self.name]
+    delete_cmd.extend(util.GetDefaultGcloudFlags(self))
+    vm_util.IssueCommand(delete_cmd)
+
+  def _Exists(self):
+    """Returns True if the Network resource exists."""
+    describe_cmd = [FLAGS.gcloud_path,
+                    'compute',
+                    'networks',
+                    'describe',
+                    self.name]
+    describe_cmd.extend(util.GetDefaultGcloudFlags(self))
+    _, _, retcode = vm_util.IssueCommand(describe_cmd, suppress_warning=True)
+    if retcode:
+      return False
+    return True
+
+
 class GceNetwork(network.BaseNetwork):
   """Object representing a GCE Network."""
 
   CLOUD = GCP
 
+  def __init__(self, network_spec):
+    super(GceNetwork, self).__init__(network_spec)
+    self.project = network_spec.project
+    name = 'pkb-network-%s' % FLAGS.run_uri
+    self.network_resource = GceNetworkResource(name, self.project)
```

Comment:
Yeah - the GceNetwork class needs to inherit from the BaseNetwork class. The GceNetworkResource class needs to inherit from BaseResource. Making them a single class and having it inherit from both leads to problems (depending on the order of base classes, the BaseNetwork Create method may get called instead of the BaseResource one or the BaseNetwork __init__ method wont get called properly because their __init__ methods take different arguments). Rather than trying to make this work, it seemed simpler to have two separate classes.